caffe2/utils/conversions.h

```diff
@@ -0,0 +1,185 @@
+#pragma once
+
+#include <caffe2/core/types.h>
+
+#ifdef __CUDA_ARCH__
+#include <cuda_fp16.h>
+#endif
+
+#ifdef __CUDA_ARCH__
+#define CONVERSIONS_DECL __host__ __device__ inline
+#else
+#define CONVERSIONS_DECL inline
+#endif
+
+namespace caffe2 {
+
+namespace convert {
+
+namespace {
+inline float16 cpu_float2half_rn(float f) {
+    float16 ret;
+
+    unsigned x = *((int*)(void*)(&f));
```

Comment:
I don't understand how this is meant to work if I can't dereference that pointer -- I need the actual value of x as originally done.